packages/core/src/2d/sprite/SpriteRenderer.ts

```diff
@@ -0,0 +1,210 @@
+
+  /**
+   * The Sprite to render.
+   */
+  get sprite(): Sprite {
+    return this._sprite;
+  }
+
+  set sprite(value: Sprite | null) {
+    if (this._sprite !== value) {
+      this._sprite = value;
+      this._setDirtyFlagTrue(SpriteRenderer._SPRITE_FLAG);
+    }
+  }
+
+  /**
+   * Rendering color for the Sprite graphic.
+   */
+  get color(): Color {
+    return this._color;
+  }
+
+  set color(value: Color) {
+    if (this._color !== value) {
+      value.cloneTo(this._color);
+    }
+  }
+
+  /**
+   * Flips the sprite on the X axis.
+   */
+  get flipX(): boolean {
+    return this._flipX;
+  }
+
+  set flipX(value: boolean) {
+    if (this._flipX !== value) {
+      this._flipX = value;
+      this._setDirtyFlagTrue(SpriteRenderer._FLIP_X_FLAG);
+    }
+  }
+
+  /**
+   * Flips the sprite on the Y axis.
+   */
+  get flipY(): boolean {
+    return this._flipY;
+  }
+
+  set flipY(value: boolean) {
+    if (this._flipY !== value) {
+      this._flipY = value;
+      this._setDirtyFlagTrue(SpriteRenderer._FLIP_Y_FLAG);
+    }
+  }
+
+  /**
+   * Create a sprite renderer instance.
+   * @param entity - Entity to which the sprite renderer belongs
+   */
+  constructor(entity: Entity) {
+    super(entity);
+
+    this._sprite = null;
+    this._color = new Color(1, 1, 1, 1);
+    this._flipX = false;
+    this._flipY = false;
+    this._dirtyFlag = SpriteRenderer._ALL_FLAG;
+    this._isWorldMatrixDirty = entity.transform.registerWorldChangeFlag();
+  }
+
+  /**
+   * Push the render data of the sprite to render queue.
+   * @param camera - Camera which is rendering
+   */
+  _render(camera: Camera): void {
+    const { entity, sprite, _positions } = this;
+    if (!sprite) {
+      return;
+    }
+
+    const { transform } = entity;
+
+    // Update sprite data.
+    const needUpdate = sprite._updateMeshData();
+
+    if (this._isWorldMatrixDirty.flag || needUpdate || this._isContainDirtyFlag(SpriteRenderer._SPRITE_FLAG)) {
+      // Update vertices position in world space.
+      const localPositions = sprite._positions;
+      const localPosZ = transform.position.z;
+      const localVertexPos = SpriteRenderer._tempVec3;
+      const worldMatrix = transform.worldMatrix;
+      for (let i = 0; i < 4; i++) {
+        const curVertexPos = localPositions[i];
+        localVertexPos.setValue(
+          this.flipX ? -curVertexPos.x : curVertexPos.x,
+          this.flipY ? -curVertexPos.y : curVertexPos.y,
+          localPosZ
+        );
+        Vector3.transformToVec3(localVertexPos, worldMatrix, _positions[i]);
+      }
+
+      this._setDirtyFlagFalse(SpriteRenderer._FLIP_FLAG);
+      this._setDirtyFlagFalse(SpriteRenderer._SPRITE_FLAG);
+      this._isWorldMatrixDirty.flag = false;
+    } else if (this._isContainDirtyFlag(SpriteRenderer._FLIP_FLAG)) {
+      const flipX = this._isContainDirtyFlag(SpriteRenderer._FLIP_X_FLAG);
+      const flipY = this._isContainDirtyFlag(SpriteRenderer._FLIP_Y_FLAG);
```

Comment:
Here has a BUG, If developer call ```flipY = true```, then call ```flipY = false```. Im fact you do not need to re-compute,but.........